src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -695,10 +704,18 @@ static int add_root_mount(void) {
         }
 
         /* OK, we shall look for a root device, so let's wait for a root device to show up.  A udev rule will
-         * create the link for us under the right name. */
+         * create the link for us under the right name.
+         *
+         * There are two distinct names: the /dev/gpt-auto-root-nofr symlink is always created for the root
+         * partition, and the /dev/gpt-auto-root symlink is only created if factory reset mode is either not
+         * enabled or already complete. In scenarios where the root disk is reset during factory reset the
+         * latter is the link to use, otherwise the former. */
```

Comment:
'latter' and 'former' are swapped? Or do I misunderstand the motivation?

Reply:
will extend the comment a bit